engine/src/main/java/org/pentaho/di/www/GetStatusServlet.java

```diff
@@ -0,0 +1,1104 @@
+   <os_name>Windows 7</os_name>
+   <os_version>6.1</os_version>
+   <os_arch>amd64</os_arch>
+   <transstatuslist>
+   <transstatus>
+   <transname>Row generator test</transname>
+   <id>56c93d4e-96c1-4fae-92d9-d864b0779845</id>
+   <status_desc>Waiting</status_desc>
+   <error_desc/>
+   <paused>N</paused>
+   <stepstatuslist>
+   </stepstatuslist>
+   <first_log_line_nr>0</first_log_line_nr>
+   <last_log_line_nr>0</last_log_line_nr>
+   <logging_string>&#x3c;&#x21;&#x5b;CDATA&#x5b;&#x5d;&#x5d;&#x3e;</logging_string>
+   </transstatus>
+   <transstatus>
+   <transname>dummy-trans</transname>
+   <id>c56961b2-c848-49b8-abde-76c8015e29b0</id>
+   <status_desc>Stopped</status_desc>
+   <error_desc/>
+   <paused>N</paused>
+   <stepstatuslist>
+   </stepstatuslist>
+   <first_log_line_nr>0</first_log_line_nr>
+   <last_log_line_nr>0</last_log_line_nr>
+   <logging_string>&#x3c;&#x21;&#x5b;CDATA&#x5b;&#x5d;&#x5d;&#x3e;</logging_string>
+   </transstatus>
+   </transstatuslist>
+   <jobstatuslist>
+   <jobstatus>
+   <jobname>dummy_job</jobname>
+   <id>abd61143-8174-4f27-9037-6b22fbd3e229</id>
+   <status_desc>Stopped</status_desc>
+   <error_desc/>
+   <logging_string>&#x3c;&#x21;&#x5b;CDATA&#x5b;&#x5d;&#x5d;&#x3e;</logging_string>
+   <first_log_line_nr>0</first_log_line_nr>
+   <last_log_line_nr>0</last_log_line_nr>
+   </jobstatus>
+   </jobstatuslist>
+   </serverstatus>
+   </pre>
+
+   <h3>Status Codes</h3>
+   <table class="pentaho-table">
+   <tbody>
+   <tr>
+   <th>code</th>
+   <th>description</th>
+   </tr>
+   <tr>
+   <td>200</td>
+   <td>Request was processed.</td>
+   </tr>
+   <tr>
+   <td>500</td>
+   <td>Internal server error occurs during request processing.</td>
+   </tr>
+   </tbody>
+   </table>
+   </div>
+   */
+  public void doGet( HttpServletRequest request, HttpServletResponse response ) throws ServletException,
+      IOException {
+    if ( isJettyMode() && !request.getContextPath().startsWith( CONTEXT_PATH ) ) {
+      return;
+    }
+
+    if ( log.isDebug() ) {
+      logDebug( BaseMessages.getString( PKG, "GetStatusServlet.StatusRequested" ) );
+    }
+    response.setStatus( HttpServletResponse.SC_OK );
+    String root = request.getRequestURI() == null ? StatusServletUtils.PENTAHO_ROOT
+      : request.getRequestURI().substring( 0, request.getRequestURI().indexOf( CONTEXT_PATH ) );
+    String prefix = isJettyMode() ? StatusServletUtils.STATIC_PATH : root + StatusServletUtils.RESOURCES_PATH;
+    prefix = encodeUriComponents( prefix );
+    root = encodeUriComponents( root );
+    boolean useXML = "Y".equalsIgnoreCase( request.getParameter( "xml" ) );
+    boolean useLightTheme = "Y".equalsIgnoreCase( request.getParameter( "useLightTheme" ) );
+
+    if ( useXML ) {
+      response.setContentType( "text/xml" );
+      response.setCharacterEncoding( Const.XML_ENCODING );
+    } else {
+      response.setContentType( "text/html;charset=UTF-8" );
+    }
+
+    PrintWriter out = response.getWriter();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *out*. Currently, there are execution paths that do not contain closure statements, for example, when Encode.forHtml throws an exception. Ensure *out* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).